Plugins~/Src/MeshSyncClientBlender/python/unity_mesh_sync_baking.py

```diff
@@ -0,0 +1,1195 @@
+        materials = []
+        finalMaterials = []
+        for matSlot in obj.material_slots:
+            materials.append(matSlot.material)
+
+        if bakeIndividualMats:
+            for matSlot in obj.material_slots:
+                matSlot.material = None
+
+        try:
+            for matIndex, mat in enumerate(materials):
+                obj.material_slots[matIndex].material = mat
+
+                if not self.canMaterialBeBaked(mat):
+                    continue
+
+                obj.active_material_index = matIndex
+
+                matOutput, bsdf = self.findMaterialOutputNodeAndInput(mat)
+
+                if matOutput is None or bsdf is None:
+                    continue
+
+                self.bakedImageNodeYOffset = 0  # To keep track of image node location for this object
+                self.objectBakeInfo = {}  # To keep track of baked channels for this object to check if image nodes can be reused
+
+                # Ensure object is not hidden, otherwise baking will fail:
+                wasHidden = obj.hide_get()
+                obj.hide_set(False)
+                context.view_layer.objects.active = obj
+
+                self.deselectAllMaterialNodes(mat)
+
+                msb_log(f"********** Checking if '{mat.name}' on '{obj.name}' needs baked materials. **********")
+
+                # If any channel was baked, it will be on a new material,
+                # store that to frame all new nodes after everything is baked:
+                bakedMat = mat
+                for channel in BAKED_CHANNELS:
+                    if not self.isChannelBakeEnabled(context, channel):
+                        continue
+
+                    self.incrementProgress(context, f"Baking '{mat.name}'->{channel} on '{obj.name}'", obj)
+                    if bakeSettings.run_modal:
+                        yield
+                        context = self.context
+
+                    didBake, newMat = self.bakeBSDFChannelIfNeeded(context, obj, mat, bsdf, matOutput, channel)
+                    if didBake:
+                        bakedMat = newMat
+
+                    if bakeSettings.run_modal:
+                        yield
+                        context = self.context
+
+                if bakedMat != mat:
+                    self.cleanUpNodeTreeAndConnectBakedBSDF(bakedMat, matOutput)
+
+                # Needed for restore afterwards:
+                finalMaterials.append(bakedMat)
+
+                obj.material_slots[matIndex].material = None
+
+                # Restore state:
+                obj.select_set(False)
+                obj.hide_set(wasHidden)
+
+                if bakeSettings.run_modal:
+                    yield
+                    context = self.context
+
+            if UV_OVERRIDE in obj.data and len(obj.data.uv_layers) > 1:
+                msb_log(
+                    f"New UVs were generated for '{obj.name}' for baking. Old UVs need to be deleted so the baked textures work correctly.")
```

Comment:
Could this delete a user's UVs accidentally? Can the user specify if they want their UVs overridden?

Reply:
Yes, UVs can be lost or overwritten. The user can choose whether to generate UVs and risk that or not.

Reply:
Does the user know their UVs will be lost/overwritten?